internal/config/config.go

```diff
@@ -618,110 +518,81 @@ func (c *Config) GetProvider() cloudprovider.Provider {
+		return err
 	}
-	if strings.Contains(userInput, ",") {
-		return false
+
+	// Register Provider validation error types
+	if err := validate.RegisterTranslation("no_provider", trans, registerNoProviderError, translateNoProviderError); err != nil {
+		return err
 	}
-	if strings.Contains(userInput, ";") {
-		return false
+
+	if err := validate.RegisterTranslation("more_than_one_provider", trans, registerMoreThanOneProviderError, c.translateMoreThanOneProviderError); err != nil {
+		return err
 	}
 
-	splitInstanceType := strings.Split(userInput, ".")
+	// register custom validator with label supported_k8s_version to validate version based on available versionConfigs.
+	if err := validate.RegisterValidation("supported_k8s_version", validateK8sVersion); err != nil {
+		return err
+	}
 
-	if len(splitInstanceType) != 2 {
-		return false
+	// register custom validator with label aws_instance_type to validate the AWS instance type from config input.
+	if err := validate.RegisterValidation("aws_instance_type", validateAWSInstanceType); err != nil {
+		return err
 	}
 
-	userDefinedFamily := splitInstanceType[0]
-	userDefinedSize := splitInstanceType[1]
+	// register custom validator with label azure_instance_type to validate the Azure instance type from config input.
+	if err := validate.RegisterValidation("azure_instance_type", validateAzureInstanceType); err != nil {
+		return err
+	}
 
-	// Check if instace type has at least 4 vCPUs (= contains "xlarge" in its name)
-	hasEnoughVCPUs := strings.Contains(userDefinedSize, "xlarge")
-	if !hasEnoughVCPUs {
-		return false
+	// register custom validator with label gcp_instance_type to validate the GCP instance type from config input.
+	if err := validate.RegisterValidation("gcp_instance_type", validateGCPInstanceType); err != nil {
+		return err
 	}
 
-	// Now check if the user input is a supported family
-	// Note that we cannot directly use the family split from the Graviton check above, as some instances are directly specified by their full name and not just the family in general
-	for _, supportedFamily := range instancetypes.AWSSupportedInstanceFamilies {
-		supportedFamilyLowercase := strings.ToLower(supportedFamily)
-		if userDefinedFamily == supportedFamilyLowercase {
-			return true
-		}
+	// Register provider validation
+	validate.RegisterStructValidation(validateProvider, ProviderConfig{})
+
+	err := validate.Struct(c)
+	if err == nil {
+		return nil
 	}
 
-	return false
-}
+	var errs validator.ValidationErrors
+	if !errors.As(err, &errs) {
+		return err
+	}
 
-// IsDebugCluster checks whether the cluster is configured as a debug cluster.
-func (c *Config) IsDebugCluster() bool {
-	if c.DebugCluster != nil && *c.DebugCluster {
-		return true
+	var validationErrors error
+	for _, e := range errs {
+		validationErrors = multierr.Append(
+			validationErrors,
+			errors.New(e.Translate(trans)),
+		)
```

Comment:
nit: For "seperation of concerns"-like reasons, I prefer the old variant of this function and would move the conversion to error into WithValidate(). But if you decide to keep it like this, please change the doc comment of this func.